types/vtex.js/index.d.ts

```diff
@@ -0,0 +1,865 @@
+        paymentSystem: string;
+        bin?: string;
+        paymentName?: string;
+        paymentGroupName?: string;
+        value: number;
+        installments: Installment[];
+    }
+
+    interface GiftCard {
+        redemptionCode: string;
+        value: number;
+        balance: number;
+        name: string;
+        id: string;
+        inUse: boolean;
+        isSpecialCard: boolean;
+    }
+
+    interface AvailableAccount {
+        accountId: string;
+        paymentSystem: string;
+        paymentSystemName: string;
+        cardNumber: string;
+        availableAddresses: string[];
+    }
+
+    interface PaymentData {
+        installmentOptions: InstallmentOption[];
+        paymentSystems: PaymentSystem[];
+        payments: Payment[];
+        giftCards: GiftCard[];
+        giftCardMessages: any[];
+        availableAccounts: AvailableAccount[];
+        availableTokens: any[];
+    }
+
+    interface PaymentSystem {
+        id: number;
+        name: string;
+        groupName: string;
+        validator: Validator;
+        stringId: string;
+        template: string;
+        requiresDocument: boolean;
+        selected: boolean;
+        isCustom: boolean;
+        description: string;
+    }
+
+    interface Validator {
+        regex: string;
+        mask: string;
+        cardCodeRegex: string;
+        cardCodeMask: string;
+        weights: number[];
+    }
+
+    interface ShippingData {
+        address?: CheckoutAddress;
+        logisticsInfo: LogisticsInfo[];
+        selectedAddresses: CheckoutAddress[];
+        availableAddresses: CheckoutAddress[];
+        pickupPoints: PickupPoint[];
+    }
+
+    interface PickupPoint {
+        friendlyName: string;
+        address: CheckoutAddress;
+        additionalInfo: string;
+        id: string;
+        businessHours: BusinessHour[];
+    }
+
+    interface LogisticsInfo {
+        addressId?: string;
+        deliveryChannels: DeliveryChannel[];
+        itemId: string;
+        itemIndex: number;
+        shipsTo: string[];
+        slas: SLA[];
+        selectedDeliveryChannel?: string;
+        selectedSla?: string;
+    }
+
+    interface DeliveryChannel {
+        id: "pickup-in-point" | "delivery";
+    }
+
+    interface AvailableDeliveryWindow {
+        endDateUtc: string;
+        listprice: number;
+        startDateUtc: string;
+        tax: number;
+    }
+
+    interface SLA {
+        id: string;
+        deliveryChannel: string;
+        name: string;
+        deliveryIds: DeliveryId[];
+        shippingEstimate: string;
+        shippingEstimateDate?: string;
+        lockTTL?: string;
+        availableDeliveryWindows: AvailableDeliveryWindow[];
+        deliveryWindow?: string;
+        price: number;
+        listPrice: number;
+        tax: number;
+        pickupStoreInfo: PickupStoreInfo;
+        pickupPointId?: string;
+        pickupDistance?: number;
+        polygonName?: string;
+        transitTime?: string;
```

Comment:
I highly suspect that these should all have `| undefined` to support `exactOptionalPropertyTypes`. (And other props too)

See: [Playground Link](https://www.typescriptlang.org/play?exactOptionalPropertyTypes=true&ts=4.4.2#code/PTAEHUEsBcAtQKYA8CGBjaB5ADtSB7AOxQBsAFAJ32wQugE8AVemgZ0WICMSEATAGlDMaAZTQVIuALAAoEKFgp2raBIy1QFAK492KKlsK9QAd0XRQ9fFtBoUhUKwQX7oaniKlZ87FRp16UGh8ADpZbzAAMXwKRFQAW2weQThIdkhCaFoAM3QEUF4ENBJ9BHY4FAs4Wny00BQI0F9qWgZTWEg0eDsHTnyifuygk3xHVQyAc1YALlAAcl59AGs5t1i5kkgJ2Gg5xpjQGEdYaxJjQnwLPsOHardOACsi6DCZWQysily0fIBVJwoABEELkdNB2ABvWSgUDyRiwfIoThOQg-NxDFCgABupC0+QoCF8ZQQmXYc1Y9BUCHiexkMLQ+BIMXh1IQmCxtAkhQA-LMAESLChLPmgAA+oD5m220D5AG5ZABfcJyMBQODWKoddjZEoTDhIngCIIIgn1U1wAn5HEkPHsMydeBWGw9RpOCwAAwZTIoLPibI5FC5CHdQVGfX5guFgklWx2IvsxndhkK2QyfHdr0aImceEIeru1rxodASaMILTvBDJkgJBI9Vr+BMoHi+BUoAAUigcWIJLhNIY8H7Gl0ikttQc7sg0llUflgqBcjXBPauod2CDss9IBySIFciQKa95AAJRsIAMpE0Icn1BdaABe97TrHKo1S7D6iixBAoISECIKctCBgAhCEabIDk9RlmQRP12U5SBChDAB9Xh8DKQg5mgZCbmKLRChLZMgPTTMVX-fIdRQPV3WQdAsFwUDSEoFoAmEMoQ3iFAljKIQWAQHtJBcNA0C0ChKgQXdGk46BVzuFMMhAogmiob9CmMJR6gcdxGLrZp-AYUjCmKUo71RDwHAmZx-lobNoFzKYAAoAEpZmsoEQRQMFWHlN4ZAZQg2zdez2AAXlASzoDc2zgucnygsmVgQi9GDWXgwNEPyMKBWWOVZHivNEuSn1YP9BCCKyqU4x8xoACEtAsC5oGmPKcwSpLoOK1KAyDUAwqI1NCD4aqyMoH9izuSjqNojAcHMpi-FaJg+NYd1LzqEwNMappW1YSBuAQEIgA)

Reply:
Thx, I will review this soon!